src/watch/watch.ts

```diff
@@ -64,9 +70,10 @@ export class Watcher {
 		this.buildTimeout = setTimeout(() => {
 			this.buildTimeout = null;
 			for (const id of this.invalidatedIds) {
-				this.emit('change', id);
+				this.emit('change', id, this.deletedIds.has(id));
```

Comment:
If you go with the suggestion of handling "add" here as well, one should probably think about ordering here. In the end, a file can have only one resulting change. E.g. first changing, then deleting in the same cycle would result in an "unlink" (or "delete" if you prefer). On the other hand, first adding and then changing would result in an "add". And finally first "add" and then "delete" should probably purge this id altogether and not emit an event.

So maybe instead of a set of deletedIds, maintain a Map (or object) of ids to the event that should be emitted? And then update this Map if an event happens for an id that is already in the Map:
- "add" + "unlink" -> remove from Map
- "add" + "change" -> "add"
- "change" + "unlink" -> "unlink"
- "change" + "add" -> feels buggy, but should likely be handled as "change" or we throw
- "unlink" + "add" -> "change"
- "unlink" + "change" -> definitely buggy, maybe throw or ignore

Reply:
Done!